### ISSUE TITLE
Remote function call not detecting arguments

### ISSUE THREAD

**Issue by ElenaBadilloG** [ray-project/ray#2554]
When executing a @ray.remote function, the following exception is being thrown, even though I am supplying all the arguments set in the function definition:

   `  /Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-     packages/ray/signature.py in extend_args(function_signature, args, kwargs)
      208                     raise Exception("No value was provided for the argument "
      209                                     "'{}' for the function '{}'.".format(
      --> 210                                         keyword_name, function_name))
      211 
      212     no_positionals = len(arg_is_positionals) == 0 or not arg_is_positionals[-1]

      Exception: No value was provided for the argument 'phones' for the function 'compile_file'.`

A minimal example of my remote function definition and remote call are as follows:

`    import ray
      ray.init(num_cpus=4, num_gpus=1, include_webui=False) #initialize ray with 4 CPUs

    @ray.remote
    def compile_file(self, rgx_patts, phones): # method for my Case class

        self._phones = self.phonelist(rgx_patts, phones)

    def compile_all(inputDirectory='C/TODOS', phones = ['10002000']):  
        d = {}
        file_lst = pdfLister(inputDirectory, termin)
        for i, file in enumerate(file_lst):
             doc = Case(file)
             doc.compile_file.remote(rgx_patts, phones) # exception thrown here
            d[i] = doc
        case_dic = {k: ray.get(dic_id) for k, dic_id in d.items()}
        return case_dic`

**Comment by ericl** [comment]
Hey Elena, the `@ray.remote` should applied on the `Case` class itself and not its method. After that you can do `doc = Case.remote(file)` and then `doc.compile_file.remote` after.

Alternatively, you could do `doc.compile_file.remote(doc, rgx_patts, phones)` and that might work.

**Comment by GiliR4t1qbit** [comment]
I'm new to Ray, but I'm not sure that's the best answer here, in general... If I understand it correctly, there is a big difference between a remote function and an actor, and if all you want to do is call a remote method that is stateless then I don't think it's appropriate to make the class an actor. One repercussion for making it an actor is that you will have to instantiate a specific number of actors, and each actor will be assigned to a node, and you will have to choose which actor to call when you want to call a method of the actor class... 

I have another solution: pass the object as the first argument, which will then be read as "self" and the code will work fine. Basically, replace:
doc.compile_file.remote(rgx_patts, phones)
with 
doc.compile_file.remote(doc, rgx_patts, phones)

This is the answer I added to Elena's Stackoverflow question: https://stackoverflow.com/questions/51673310/remote-ray-call-not-detecting-function-arguments/52903322#52903322, and it seems to work for my case. I'm wondering though if it would make sense for remote methods to be "smart" enough to figure out that their first argument is special and "fill it in" by themselves. Certainly the call I suggested above is strange looking and redundant (but it seems to work). 
@ericl, @robertnishihara, @ElenaBadilloG
